tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -3074,9 +3074,19 @@ ReductionCodegenInfo IrEmitterUnnested::ComputeReductionCodegenInfo(
            << " " << reduction_dimensions.dimensions[0] << " "
            << reduction_dimensions.dimensions[1] << " "
            << reduction_dimensions.dimensions[2];
+  auto get_dtype_bits = [](const HloInstruction *i) {
+    return primitive_util::BitWidth(i->shape().element_type());};
 
+  // For fusion with multiple inputs, use the smallest input dtype to
+  // select the reduction_tiling.
+  int smallest_input_dtype_bits = get_dtype_bits(first_reduce->operand(0));
+  for (xla::HloInstruction* input: unnested_hlo->operands()) {
+    smallest_input_dtype_bits = std::min(get_dtype_bits(input),
```

Comment:
Now that I look closer at this code, I don't think does what the comment claims: it iterates through inputs to the *fusion* which might have different datatypes to the reduction inputs. Maybe just easier to check the first_reduce datatype and call it a day? I don't think we even fuse different datatypes.

Reply:
*input* dtypes can be mixed. We see that in particular with mixed precision. Big tensor are in float16 while the smaller tensor that are broadcasted are in float32. I do not know if we merge mixed output dtype.

Here, some of the tiling size logic is based on dtypes. I think the code does what the comment say: find the smallest dtype on all the inputs tensor of the *fusion*. If you understood something else, maybe the comment can be made more clear? The comment do not refer to reduction at all (but it is in a function specific for reduction).